### ISSUE TITLE
Bug in Nav Drawer

### ISSUE THREAD

**Issue by nitish1211** [systers/PCSA-Android#436]
Bug: After opening a particular fragment and pressing back to got to the previous one when we try to reopen the latter one via the Nav Drawer, then the Nav Drawer simply closes without opening the required Fragment.

![g_20170312_1904489](https://cloud.githubusercontent.com/assets/13676490/23832270/6e82bd5a-0757-11e7-9c2d-a5978c6eb9bb.gif)

@sandarumk @top-gun007 Can I send a PR for this?

**Comment by codingblazer** [comment]
Hey @nitish1211 , good catch but it will be solved when #405 #414 #427 gets solved.

**Comment by nitish1211** [comment]
Thanks , I think the entire code for Nav drawer navigation needs to be completely remodelled.
Are you currently working on it @codingblazer .
> #414 Kinda Explains the reason for the bug.

**Comment by codingblazer** [comment]
@nitish1211 
Yes, it needed a great amount of modification. I am done with most of it :-)
